libs/wallet/src/use-vega-transaction-store.tsx

```diff
@@ -0,0 +1,168 @@
+import produce from 'immer';
+import type { Transaction } from './connectors';
+import {
+  isWithdrawTransaction,
+  isOrderSubmissionTransaction,
+  isOrderCancellationTransaction,
+  isOrderAmendmentTransaction,
+} from './connectors';
+import { determineId } from './utils';
+
+import create from 'zustand';
+import type { VegaTxState } from './use-vega-transaction';
+import { VegaTxStatus } from './use-vega-transaction';
+import type {
+  TransactionEventFieldsFragment,
+  WithdrawalBusEventFieldsFragment,
+  OrderBusEventFieldsFragment,
+} from './__generated__/TransactionResult';
+
+import type { WithdrawalApprovalQuery } from './__generated__/WithdrawalApproval';
+export interface VegaStoredTxState extends VegaTxState {
+  id: number;
+  createdAt: Date;
+  updatedAt: Date;
+  body: Transaction;
+  transactionResult?: TransactionEventFieldsFragment;
+  withdrawal?: WithdrawalBusEventFieldsFragment;
+  withdrawalApproval?: WithdrawalApprovalQuery['erc20WithdrawalApproval'];
+  order?: OrderBusEventFieldsFragment;
+}
+export interface VegaTransactionStore {
+  transactions: (VegaStoredTxState | undefined)[];
+  create: (tx: Transaction) => number;
+  update: (
+    index: number,
+    update: Partial<
+      Pick<VegaStoredTxState, 'status' | 'txHash' | 'signature' | 'error'>
+    >
+  ) => void;
+  dismiss: (index: number) => void;
+  delete: (index: number) => void;
+  updateWithdrawal: (
+    withdrawal: NonNullable<VegaStoredTxState['withdrawal']>,
+    withdrawalApproval: NonNullable<VegaStoredTxState['withdrawalApproval']>
+  ) => void;
+  updateOrder: (order: OrderBusEventFieldsFragment) => void;
+  updateTransactionResult: (
+    transactionResult: TransactionEventFieldsFragment
+  ) => void;
+}
+
+export const useVegaTransactionStore = create<VegaTransactionStore>(
+  (set, get) => ({
+    transactions: [] as VegaStoredTxState[],
+    create: (body: Transaction) => {
+      const transactions = get().transactions;
+      const now = new Date();
+      const transaction: VegaStoredTxState = {
+        id: transactions.length,
+        createdAt: now,
+        updatedAt: now,
+        body,
+        error: null,
+        txHash: null,
+        signature: null,
+        status: VegaTxStatus.Requested,
+        dialogOpen: true,
```

Comment:
question (non blocking): why do we need to track dialogOpen state?

Reply:
dialogOpen indicates it dialog connected with stored transaction should be visible to the user. It controls with toast notification is open. Will be set to false if user will dismiss it. 
It was not renamed to keep a consistency.